torch_np/_ufunc_impl.py

```diff
@@ -0,0 +1,152 @@
+import torch
+
+from . import _util
+from . import _helpers
+
+def deco_binary_ufunc(torch_func):
+    """Common infra for binary ufuncs: receive arrays, sort out type casting,
+       broadcasting, out array handling etc, and delegate to the
+       pytorch function for actual work, then wrap the results into an array.
+
+       x1, x2 are arrays! array_like -> array conversion is the caller responsibility.
+    """
+    def wrapped(x1, x2, /, out=None, *, where=True,
+                casting='same_kind', order='K', dtype=None, subok=False, **kwds):
+        _util.subok_not_ok(subok=subok)
+        if order != 'K' or not where:
+            raise NotImplementedError
+
+        # XXX: dtype=... parameter is silently ignored
```

Comment:
Let's either assert that `dtype is None` for now. But more generally, why not implementing it?

Reply:
Certainly will. The next batch of numpy tests will likely make me bite the bullet and figure out all the weak promotion rules, dtype resolution rules and how that interacts with registered loop types.

Reply:
Let's throw in an assert for now though?

Reply:
https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/17/commits/2b363f3a4d2e0d994608872e6e84e261e53b5d5d